crates/miden-objects/src/asset/vault/mod.rs

```diff
@@ -291,12 +298,16 @@ impl AssetVault {
     ///
     /// # Errors
     /// - The non-fungible asset is not found in the vault.
+    /// - The maximum number of leaves per asset is exceeded.
```

Comment:
For consistency, can you add this line to the other `AssetVault` functions where we've introduced the `AssetVaultError::MaxLeafEntriesExceeded`?